src/components/bottom-widget.js

```diff
@@ -97,7 +97,7 @@ export default function BottomWidgetFactory(TimeWidget, AnimationControl) {
             updateAnimationSpeed={visStateActions.updateLayerAnimationSpeed}
           />
         ) : null}
-        {enlargedFilterIdx > -1 ? (
+        {enlargedFilterIdx > -1 && Object.keys(datasets).length > 0 ? (
```

Comment:
this doesn't make sense, how come there is filter but no dataset? what could have happened is when deleting a dataset, the associated filter was not deleted

Reply:
@heshan0131 you are right. In fact the root cause was related to the filter not being deleted correctly which I fixed https://github.com/keplergl/kepler.gl/pull/835/files#diff-3c42abf857bb61c5f6578c72771405d2R910

I put this as an extra check

Reply:
Oh nice. Could you add a test case for this?